frontend/services/src/users/usersService.ts

```diff
@@ -0,0 +1,109 @@
+import { NewUserData, UserData, UserId, PublicUserData, PrivateUserData, EmptyUserData } from "@/interfaces/UserTypes";
+import { addDoc, collection, doc, getDoc, getDocs, deleteDoc, updateDoc, setDoc } from "firebase/firestore";
+import { db } from "../firebase";
+import { extractPrivateUserData, extractPublicUserData } from "./usersUtils/createUsersUtils";
+
+export async function createUser(data: NewUserData, userId: string) {
+  try {
+    console.log("triggered");
+    console.log(data, userId);
+    // const docRef = await setDoc(doc(db, "Users", userId), data);
+    const publicDocRef = await setDoc(doc(db, "Users", "Active", "Public", userId), extractPublicUserData(data));
+    const privateDocRef = await setDoc(doc(db, "Users", "Active", "Private", userId), extractPrivateUserData(data));
+    // return docRef.id;
+  } catch (error) {
+    console.error(error);
+    throw error;
+  }
+}
+
+export async function getPublicUserById(userId: UserId): Promise<UserData> {
+  console.log(userId);
+  if (userId === undefined) {
+    throw Error;
+  }
+  try {
+    const userDoc = await getDoc(doc(db, "Users", "Active", "Public", userId));
+    if (!userDoc.exists()) {
+      return EmptyUserData;
+    }
+    console.log("getid", userId);
+    const userData = userDoc.data() as UserData;
+    userData.userId = userId;
+    return userData;
+  } catch (error) {
+    console.error(error);
+    throw error;
+  }
+}
+
+export async function getFullUserById(userId: UserId): Promise<UserData> {
+  console.log(userId);
+  if (userId === undefined) {
+    throw new Error("User ID is undefined");
+  }
+  try {
+    const publicDoc = await getDoc(doc(db, "Users", "Active", "Public", userId));
+    const privateDoc = await getDoc(doc(db, "Users", "Active", "Private", userId));
+
+    if (!publicDoc.exists() && !privateDoc.exists()) {
```

Comment:
what is the thought process behind this? would we ever have a situation where only one document exists? possibly just after signup where they still haven't filled in their information. would this line fail early or continue?